platform/security/secure-boot/src/test/groovy/org/codice/ddf/security/boot/SecureBootSpec.groovy

```diff
@@ -0,0 +1,258 @@
+            boolean securityManagerEnabled() {
+                return null;
+            }
+            void systemExit(Exception e) {
+                exitedDueToSystemServiceFailureToExit = true
+            }
+        }
+        secureBoot.init()
+
+        then:
+        0 * systemService.halt("0")
+        exitedDueToSystemServiceFailureToExit == false
+    }
+
+    def 'SystemService throws exception when asked to halt'() {
+        setup:
+        def exitedDueToSystemServiceFailureToExit = false
+        def userHomeSysProp = userHome.getRoot().toString()
+        def ddfHomeSysProp = userHome.newFolder("ddf").toString()
+        def systemService = Mock(SystemService)
+        def securityManager = Mock(SecurityManager)
+        System.properties.'ddf.home' = ddfHomeSysProp
+        System.properties.'user.home' = userHomeSysProp
+
+        when:
+        def secureBoot = new SecureBoot(systemService) {
+            boolean securityManagerEnabled() {
+                return securityManager;
+            }
+
+            void systemExit(Exception e) {
+                exitedDueToSystemServiceFailureToExit = true
+            }
+        }
+        secureBoot.init()
+
+        then:
+        1 * systemService.halt("0") >> { throw new Exception("unable to halt")}
+        exitedDueToSystemServiceFailureToExit == true
+    }
+
+    def 'System exits normally when attempting to read invalid ddf.home'() {
+        setup:
+        def exitedDueToSystemServiceFailureToExit = false
+        def userHomeSysProp = userHome.getRoot().toString()
+        def ddfHomeSysProp = "/" + System.currentTimeMillis() + "/" + UUID.randomUUID() + "/ddf"
+        def systemService = Mock(SystemService)
+        def securityManager = Mock(SecurityManager)
+        System.properties.'ddf.home' = ddfHomeSysProp
+        System.properties.'user.home' = userHomeSysProp
+
+        when:
+        def secureBoot = new SecureBoot(systemService) {
+            boolean securityManagerEnabled() {
+                return securityManager;
+            }
+
+            void systemExit(Exception e) {
+                exitedDueToSystemServiceFailureToExit = true
+            }
+        }
+        secureBoot.init()
+
+        then:
+        1 * systemService.halt("0")
+        exitedDueToSystemServiceFailureToExit == false
+    }
+
+    /**
+     * ie. /opt/ddfHomeSymLink -> /users/tomPenny/ddf
+     */
+    def 'System exits normally when security manager enabled and symbolic link outside of user home dir pointing to ddf installation inside of user home dir'() {
+        setup:
+        def exitedDueToSystemServiceFailureToExit = false
+        def optDir = temporaryFolder.newFolder("opt").toPath()
+        def tomPennyHomeDir = temporaryFolder.newFolder("tomPenny")
+        def ddfHomeDir = Files.createDirectory(Paths.get(tomPennyHomeDir.toString(), "ddf"))
+        def ddfHomeSymLink = Files.createSymbolicLink(Paths.get(optDir.toString(), "ddfHomeSymLink"), ddfHomeDir)
```

Comment:
❓ Does it work on WIndows? symlinks testing on Windows are always a big pain.

Reply:
I have tested these on Windows and they work.